src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,59 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
+    private static final String SPECIFIED_STRING_S = "supply";
+    private static final String SPECIFIED_STRING_B = "buy";
+
     public void getStatistic(String fromFileName, String toFileName) {
+        try {
+            String data = readFile(fromFileName);
+            String report = generateReport(data);
+            writeToFile(toFileName, report);
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private String readFile(String fromFileName) throws IOException {
+        StringBuilder data = new StringBuilder();
+        try (BufferedReader bufferedReader = new BufferedReader(new FileReader(fromFileName))) {
+            String line;
+            while ((line = bufferedReader.readLine()) != null) {
+                data.append(line).append("\n");
+            }
+        }
+        return data.toString();
+    }
+
+    private String generateReport(String data) {
+        int buyTotal = 0;
+        int supplyTotal = 0;
+        String[] lines = data.split("\n");
```

Comment:
It is better to use the `System.lineSeparator()`
```suggestion
        String[] lines = data.split(System.lineSeparator());
```